Makefile

```diff
@@ -1,33 +1,69 @@
 PREFIX=/usr/local
 
-CFLAGS=-O3 -std=c99 -Wall -Wextra -g -fPIC -I.
+CFLAGS  := -O3 -std=c99 -Wall -Wextra -g -fPIC -I. $(CPPFLAGS)
+LDFLAGS +=
+CC := $(CROSS)gcc
+AR := $(CROSS)ar
 
-all: libobuparse.so
+ifneq (,$(findstring mingw,$(CC)))
+	LIBSUF=.dll
+	EXESUF=.exe
+	SYSTEM=MINGW
+else
+	LIBSUF=.so
+	LDFLAGS=-Wl,--version-script,obuparse.v
+endif
+
+all: libobuparse$(LIBSUF) libobuparse.a
 
 clean:
-	@rm -fv *.so *.o
-	@rm -fv tools/obudump tools/*.o
+	@rm -fv *.so *.o *.a *.dll
+	@rm -fv tools/obudump$(EXESUF) tools/*.o
+
+libobuparse.a: obuparse.o
+	$(AR) rcs $@ $^
+
+libobuparse$(LIBSUF): obuparse.o
+	$(CC) $(LDFLAGS) -shared $^ -o $@
 
-libobuparse.so: obuparse.o
-	$(CC) -Wl,--version-script,obuparse.v -shared $^ -o $@
+install: install-shared install-static
+
+install-shared: libobuparse$(LIBSUF)
+	@install -d $(PREFIX)/include
+	@install -d $(PREFIX)/lib
+	@install -v obuparse.h $(PREFIX)/include
```

Comment:
Header should probably be split into `install-header`, otherwise you get:
```
'obuparse.h' -> '/usr/local/include/obuparse.h'
'libobuparse.so' -> '/usr/local/lib/libobuparse.so.1'
'/usr/local/lib/libobuparse.so' -> 'libobuparse.so.1'
removed '/usr/local/include/obuparse.h'
'obuparse.h' -> '/usr/local/include/obuparse.h'
'libobuparse.a' -> '/usr/local/lib/libobuparse.a'
```

Reply:
Done.  Added both an `install-header` target, as well as file checks inside of `install-shared` and `install-static` so that it should only install the header once.

Reply:
Sorry, I meant that header install should be entirely factored out and `install-shared` and `install-static` should depend on `install-header`, thus having it only run once.

Reply:
Done.